main.tf

```diff
@@ -490,12 +493,29 @@ locals {
   lbs_associated_with_cluster = length(var.additional_lb_security_group_ids) > 0 ? [for lb in data.ibm_is_lbs.all_lbs[0].load_balancers : lb.id if strcontains(lb.name, local.cluster_id)] : []
 }
 
+data "ibm_is_vpc" "vpc" {
+  identifier = var.vpc_id
+}
+
+resource "ibm_is_virtual_endpoint_gateway" "private_is_vpe" {
```

Comment:
I don't think the module should create the VPE here. The VPEs already exist in the environment in which the schematics agent is running. The whole effort is to go through those existing VPE rather than creating new ones.